nextflow_aws_batch_trigger.py

```diff
@@ -0,0 +1,298 @@
+        )
+
+
+def create_queue(batch_client, job_queue_name, compute_environment_name):
+    response = batch_client.describe_job_queues()
+    queue_names = [queue["jobQueueName"] for queue in response["jobQueues"]]
+    if job_queue_name not in queue_names:
+        response = batch_client.create_job_queue(
+            jobQueueName=job_queue_name,
+            state="ENABLED",
+            priority=1,
+            computeEnvironmentOrder=[
+                {"order": 100, "computeEnvironment": compute_environment_name},
+            ],
+        )
+        logging.info(f"The queue: {job_queue_name} has been created.")
+    else:
+        logging.info(
+            f"The Queue : {job_queue_name} has already been created. "
+            f"Hence, moving to the next process"
+        )
+
+
+def create_instance(
+    ec2_client,
+    key_name,
+    instance_role,
+    s3_data,
+    s3_bucket,
+    result_location,
+    script_name,
+    config_file_name,
+    s3_logging_dir,
+    s3_result,
+    subnet1,
+    security_groupId,
+):
+    instances = ec2_client.run_instances(
+        ImageId="ami-0b0dcb5067f052a63",
+        MinCount=1,
+        MaxCount=1,
+        InstanceType="t2.micro",
+        KeyName=key_name,
+        IamInstanceProfile={"Name": instance_role},
+        UserData=f"""#!/bin/bash 
+                 sudo rpm --import https://yum.corretto.aws/corretto.key 
+                 sudo curl -L -o /etc/yum.repos.d/corretto.repo https://yum.corretto.aws/corretto.repo 
+                 sudo yum install -y  java-11-amazon-corretto-devel 
+                 sudo amazon-linux-extras install epel -y
+                 sudo yum install s3fs-fuse -y 
+                 aws s3 sync {s3_data} /home/ec2-user/
+                 sudo chmod 777 /home/ec2-user/nextflow 
+                 cd /home/ec2-user
+                 mkdir {result_location}
+                 sudo s3fs {s3_bucket} {result_location} -o allow_other -o umask=000 -o iam_role=auto 
+                 ./nextflow run /home/ec2-user/{script_name} -c {config_file_name} -bucket-dir {s3_logging_dir} --outdir={s3_result} 
+                 cd /home/ec2-user/ 
+                 touch done.txt
+                 aws s3 cp /home/ec2-user/done.txt s3://{s3_bucket}/{result_location}""",
+        NetworkInterfaces=[
+            {
+                "AssociatePublicIpAddress": True,
+                "DeleteOnTermination": True,
+                "DeviceIndex": 0,
+                "SubnetId": subnet1,
+                "Groups": [security_groupId],
+            },
+        ],
+    )
+    logging.info(
+        f"The instance: {instances['Instances'][0]['InstanceId']} has been created and the nextflow script {script_name} has been triggered "
+    )
+
+    return instances["Instances"][0]["InstanceId"]
+
+
+def check_result(ec2_client, s3, s3_bucket, id_instance, result_location):
+    bucket_name = s3_bucket
+    file_path = result_location + "done.txt"
+
+    timeout = 3600  # in seconds
```

Comment:
Lets set this as a global variable in the file. 
Is this a reasonable amount of time that we're allowing for the job to complete? Looks like its 1 hour.

Reply:
Have set it as global variable with 7 hours duration now. Hence, resolving.